### ISSUE TITLE
False-positive "async is reserved keyword in Edition 2018" when the identifier is a part of macro body

### ISSUE THREAD

**Issue by vlad20012** [intellij-rust/intellij-rust#3191]
edition = "2018"

![image](https://user-images.githubusercontent.com/3221931/50440544-c53c5f80-0907-11e9-9c17-17eb4c6deeb0.png)

```rust
macro_rules! foobar {
    ($i:tt) => { stringify!($i) };
}

fn main() {
    println!("{}", foobar!(async));
}
```

Macro def/call bodies should not be analyzed

**Comment by imkow** [comment]
Like my #3189
